src/matcher/plugin.py

```diff
@@ -336,4 +342,48 @@ def pytest_addoption(parser) -> None:
       , default=False
       )
 
+
+@pytest.hookimpl(trylast=True)
+def pytest_configure(config: pytest.Config):
+    """Register additional configuration."""
+    if config.getoption('--pm-reveal-unused-files'):
+        config.option.collectonly = True
+        reporter = UnusedFilesReporter()
+        config.pluginmanager.unregister(name='terminalreporter')
+        config.pluginmanager.register(reporter, 'terminalreporter')
+
+
+class UnusedFilesReporter:
+    """A reporter to reveal unused pattern files."""
+    def pytest_collection_finish(self, session: pytest.Session) -> None:
+        """Once test items have been collected, check for and show unused files."""
+        if not session.items:
+            return
+
+        for alg in [_try_cli_option, _try_ini_option]:
+            patterns_base_dir, _ = alg(session.items[0]._request)  # NOQA: SLF001
+            if patterns_base_dir:
+                break
```

Comment:
This code looks copy-n-pasted from another place. I'd be glad if you could make a reusable function out of it, please. Or I can do it later after we merge your work. %)

Reply:
What has puzzled me here is `_try_module_path()`. I’ve omitted it for now because as far as I understand, according to documentation, in the future the pattern files should reflect the layout (packages, modules, classes) of test files by creating the corresponding directories. 

What would the base directories look like in such a case? Each module directory has its own `/data/expected` directory?